### PR TITLE
Use uv instead of pip

### PR DESCRIPTION
**Pull request by NickM-27**
## Type of change

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [ ] Documentation Update

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] The code has been formatted using Ruff (`ruff format frigate`)

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -73,8 +73,8 @@ jobs:
           python-version: ${{ env.DEFAULT_PYTHON }}
       - name: Install requirements
         run: |
-          python3 -m pip install -U pip
-          python3 -m pip install -r docker/main/requirements-dev.txt
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          uv pip install --system -r docker/main/requirements-dev.txt
```
Comment (felipecrs) [inline .github/workflows/pull_request.yml:77]: I'd suggest you install uv with:

```
- name: Install uv
  uses: astral-sh/setup-uv@v3
```

https://docs.astral.sh/uv/guides/integration/github/

**Comment by NickM-27** [comment]
@felipecrs any ideas? Doesn't seem to be making things faster

**Comment by felipecrs** [comment]
Maybe the key to get the speed bump is having the `uv.lock` file in git.

I have to dig deeper though, I'm not so experienced with this. It's just a guess.

**Comment by xconverge** [comment]
> Maybe the key to get the speed bump is having the `uv.lock` file in git.
> 
> I have to dig deeper though, I'm not so experienced with this. It's just a guess.

I had an absolutely massive speed increase with just switching from `pip install -r requirements.txt` to `uv pip install -r requirements.txt`, so the lock file is not essential to getting the bump. I am not familiar enough with this project so let me peek at the logs but there aren't too many knobs to turn

File: docker/main/Dockerfile [hunk 2]
```diff
@@ -167,17 +166,25 @@ RUN apt-get -qq update \
...
 COPY docker/main/requirements.txt /requirements.txt
-RUN pip3 install -r /requirements.txt
+RUN uv pip install --system --no-build-isolation -r /requirements.txt
 
 # Build pysqlite3 from source
 COPY docker/main/build_pysqlite3.sh /build_pysqlite3.sh
 RUN /build_pysqlite3.sh
 
 COPY docker/main/requirements-wheels.txt /requirements-wheels.txt
+# we still have to use pip to build wheel
```
Comment (xconverge) [inline docker/main/Dockerfile:187]: there are so many dependencies in requirements-wheels.txt, why is it necessary to use `pip` instead of `uv pip` here?

Comment (xconverge) [reply docker/main/Dockerfile:187]: agh I see, https://github.com/astral-sh/uv/issues/1681

**Pull request closed without merging**